app/code/Magento/ImportExport/Model/ResourceModel/Export/AttributeGridCollection.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\ImportExport\Model\ResourceModel\Export;
+
+use Magento\Framework\Data\Collection;
+use Magento\Framework\Phrase;
+
+/**
+ * Association of attributes for grid
+ */
+class AttributeGridCollection extends Collection
+{
+    private const FILTERED_FLAG_NAME = 'agc_filtered';
+
+    /**
+     * Adding item to collection
+     *
+     * @param array $items
+     * @return $this
+     */
+    public function setItems(array $items): self
+    {
+        foreach ($items as $item) {
+            $this->addItem($item);
+        }
+
+        return $this;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getSize(): int
+    {
+        return count($this->getItems());
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function addFieldToFilter($field, $condition)
+    {
+        if (isset($condition['like'])) {
+            $value = trim((string)$condition['like'], "'%");
+            $value = str_replace('\\','',$value);
```

Comment:
That is not enough/ Please refer to \Magento\Framework\DB\Helper\AbstractHelper::escapeLikeValue that used to escape values for like condition
You should create separate private method unescapeLikeCondition($condition) where do all the job
1) Trim ' and % as you did
2) Replace \_ to _
3) Replace \% to %